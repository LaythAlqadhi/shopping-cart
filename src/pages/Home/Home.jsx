import useFakeStoreAPI from '../../hooks/useFakeStoreAPI';
import headerImg from '../../assets/header.jpg';
import css from './Home.module.css';

const Home = () => {
  return (
    <div className={css.container}>
      <h1 className="logo">eShop</h1>
      <div className={css.innerContainer}>
        <p>Style that Speaks, Deals that Delight</p>
        <p>Your journey to fabulous finds begins here.<br />Happy Shopping!</p>
      </div>
      <img
        src={headerImg}
        alt=""
        className={css.img}
        aria-hidden="true"
      />
    </div>
  );
}

export default Home;