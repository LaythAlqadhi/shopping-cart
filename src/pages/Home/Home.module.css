.container {
  margin-block: 25px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.innerContainer {
  margin-block: 25px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  text-align: center;
}

.innerContainer p:nth-child(1) {
  font-size: 1.35rem;
  font-weight: 900;
}

.img {
  max-width: 600px;
  width: 100%;
}