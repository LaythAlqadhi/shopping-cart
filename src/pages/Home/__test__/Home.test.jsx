import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import Home from '../Home';

describe('Home component', () => {
  it('renders loading element correctly', () => {
    const { container } = render(<Home />)

    expect(container).toMatchSnapshot();
  })
})