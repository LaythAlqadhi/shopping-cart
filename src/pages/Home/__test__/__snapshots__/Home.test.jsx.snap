// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home component > renders loading element correctly 1`] = `
<div>
  <div
    class="_container_da09e4"
  >
    <h1
      class="logo"
    >
      eShop
    </h1>
    <div
      class="_innerContainer_da09e4"
    >
      <p>
        Style that Speaks, Deals that Delight
      </p>
      <p>
        Your journey to fabulous finds begins here.
        <br />
        Happy Shopping!
      </p>
    </div>
    <img
      alt=""
      aria-hidden="true"
      class="_img_da09e4"
      src="/src/assets/header.jpg"
    />
  </div>
</div>
`;
